wyvern/models/guilds.py

```diff
@@ -0,0 +1,187 @@
+
+import attrs
+
+from .base import DiscordObject, Snowflake
+
+if typing.TYPE_CHECKING:
+    from .channels import GuildChannel
+    from .emojis import CustomEmoji
+    from .roles import Role
+    from .stickers import Sticker
+    from .users import User
+
+
+__all__: tuple[str, ...] = (
+    "Guild",
+    "GuildPreview",
+    "GuildWidget",
+    "GuildWidgetSettings",
+    "WelcomeScreen",
+    "WelcomeScreenChannel",
+)
+
+
+@attrs.define(kw_only=True, slots=True, repr=True)
+class WelcomeScreenChannel(DiscordObject):
+    raw: dict[str, typing.Any]
+    channel_id: Snowflake
+    description: str
+    emoji_id: typing.Optional[Snowflake]
+    emoji_name: typing.Optional[str]
+
+
+@attrs.define(kw_only=True, slots=True, repr=True)
+class WelcomeScreen(DiscordObject):
+    raw: dict[str, typing.Any]
+    description: str
+    welcome_channels: list[WelcomeScreenChannel]
+
+    @property
+    def channels(self) -> list[WelcomeScreenChannel]:
+        return self.welcome_channels
+
+
+@attrs.define(kw_only=True, slots=True, repr=True)
+class Guild(DiscordObject):
+    raw: dict[str, typing.Any]
+    id: Snowflake
+    name: str
+    icon: typing.Optional[str]
+    splash: typing.Optional[str]
+    discovery_splash: typing.Optional[str]
+    owner: bool
+    owner_id: Snowflake
+    permissions: str
+    region: str
+    afk_channel_id: typing.Optional[Snowflake]
+    afk_timeout: int
+    widget_enabled: bool
+    widget_channel_id: typing.Optional[Snowflake]
+    verification_level: int
+    default_message_notifications: int
+    explicit_content_filter: int
+    roles: list[Role]
+    emojis: list[CustomEmoji]
+    features: list[str]
+    mfa_level: int
+    application_id: typing.Optional[Snowflake]
+    system_channel_id: typing.Optional[Snowflake]
+    system_channel_flags: int
+    rules_channel_id: typing.Optional[Snowflake]
+    max_presences: typing.Optional[int]
+    max_members: int
+    vanity_url_code: typing.Optional[str]
+    description: typing.Optional[str]
+    banner: typing.Optional[str]
+    premium_tier: int
+    premium_subscription_count: int
+    preferred_locale: str
+    public_updates_channel_id: typing.Optional[Snowflake]
+    max_video_channel_users: int
+    approximate_member_count: int
+    approximate_presence_count: int
+    welcome_screen: typing.Optional[WelcomeScreen]
+    nsfw_level: int
+    stickers: list[Sticker]
+    premium_progress_bar_enabled: bool
+
+    @property
+    def icon_url(self) -> str:
+        return f"https://cdn.discordapp.com/icons/{self.id}/{self.icon}.png"
+
+    @property
+    def splash_url(self) -> str:
+        return f"https://cdn.discordapp.com/splashes/{self.id}/{self.splash}.png"
+
+    @property
+    def discovery_splash_url(self) -> str:
+        return f"https://cdn.discordapp.com/discovery-splashes/{self.id}/{self.discovery_splash}.png"
+
+    @property
+    def banner_url(self) -> str:
+        return f"https://cdn.discordapp.com/banners/{self.id}/{self.banner}.png"
```

Comment:
all of these urls are invalid when their derivative attribute is None + has to be implemented with classes just like the avatar implementation ( check files.py )